core/src/main/java/com/scalar/db/storage/jdbc/JdbcService.java

```diff
@@ -84,11 +85,19 @@ public Optional<Result> get(Get get, Connection connection)
   @SuppressFBWarnings("OBL_UNSATISFIED_OBLIGATION_EXCEPTION_EDGE")
   public Scanner getScanner(Scan scan, Connection connection)
       throws SQLException, ExecutionException {
-    operationChecker.check(scan);
+    if (scan instanceof ScanAll) {
+      operationChecker.check((ScanAll) scan);
+    } else {
+      operationChecker.check(scan);
+    }
+
     TableMetadata tableMetadata = tableMetadataManager.getTableMetadata(scan);
     ScalarDbUtils.addProjectionsForKeys(scan, tableMetadata);
 
-    SelectQuery selectQuery = buildSelectQueryForScan(scan, tableMetadata);
+    SelectQuery selectQuery =
+        scan instanceof ScanAll
+            ? buildSelectQueryForScanAll((ScanAll) scan, tableMetadata)
+            : buildSelectQueryForScan(scan, tableMetadata);
```

Comment:
I thought it was more appropriate to make the change for ScanAll in `scan()` later on inside a PR related to the transaction implementation.

Reply:
Maybe I think we can do the change for `scan()` in this PR so that we can purely change the transaction side in the PR related to the transaction implementation. What do you think?

Reply:
Understood 👍